### DISCUSSION TITLE
Installation in air gapped system

### DISCUSSION THREAD

**Discussion by old-syniex** [vmware-tanzu/community-edition#3551] [Questions and Answers]
Is there a guide how to install tanzu community edition in air gapped vshpree?

Is it eve possible?

**Comment by stmcginnis** [comment]
Hi @Syniex! We don't fully support airgapped deployment yet. You can subscribe to this issue to get updates on progress: https://github.com/vmware-tanzu/community-edition/issues/460

It's something we do want to get to soon, but it hasn't been targeted to a specific milestone yet. We do think it is important though!

**Reply by old-syniex** [reply]
@stmcginnis thank you for the response. 

Do you think it will happen soon?

**Reply by stmcginnis** [reply]
It's really hard to say at this point. The best bet will be to keep an eye on the linked issue for any updates. Thanks!

**Reply by steviestainz** [reply]
I've managed to build an air gapped TCE 0.12 installation using command line based upon an approach detailed for TKG [here](https://docs.vmware.com/en/VMware-Tanzu-Kubernetes-Grid/1.5/vmware-tanzu-kubernetes-grid-15/GUID-mgmt-clusters-airgapped-environments.html), and have used it successfully to deploy both management and worker TCE clusters (using the BOM which gets auto-created by the TCE CLI) to pull and publish the images to a local harbor instance. I don't know if this results in an acceptable deployment of TCE as far as the maintainers are concerned, but it works reliably for me without generating any obvious errors. Once clusters are built however, adding and synchronising the tce-repo repository (e.g. projects.registry.VMware.com/tce/main:0.x.x) appears to require internet access from the kapp-controller - so this needs further work on my part to understand the options.

**Reply by steviestainz** [reply]
After testing I think it's possible to use `imgpkg copy -b projects.registry.vmware.com/tce/main:0.12.0 --to-repo myrepopath/tce/packages/standard/main` in order to provision the tce-repo to an air gapped registry, but the bundle itself contains a lot of references to images located on index.docker.io and quay.io which means that you can easily hit the  docker hub rate limit when copying these elements (>9GB). It would be much better if the package images were located on a registry without download limits, as you would have to provide credentials for multiple registries simultaneously when copying just this single bundle.
